src/sidebars/search/search-controller.js

```diff
@@ -19,15 +19,23 @@
             this._watch();
         }
 
+        hasStore() {
```

Comment:
I agree that it shouldn't change the state. Either a `hasStore` and it doesn't open the store (though I would expect it to check `if (!this.store && !window.storeService)`?), or a `getStore` as you suggested.